spectral_cube/spectral_cube.py

```diff
@@ -1264,14 +1265,38 @@ def _pix_cen(self):
         x[:, 1:] = np.cumsum(np.degrees(dx), axis=1)
         y[1:, :] = np.cumsum(np.degrees(dy), axis=0)
 
-        x = x.reshape(1, x.shape[0], x.shape[1])
-        y = y.reshape(1, y.shape[0], y.shape[1])
-        spectral = spectral.reshape(-1, 1, 1) - spectral.ravel()[0]
-        x, y, spectral = np.broadcast_arrays(x, y, spectral)
+        x, y, spectral = np.broadcast_arrays(x[None,:,:], y[None,:,:], spectral[:,None,None])
 
         return spectral, y, x
 
     @cached
+    def _pix_size_slice(self, axis):
+        """
+        Return the size of each pixel along any given direction.  Assumes
+        pixels have equal size.  Also assumes that the spectral and spatial
+        directions are separable, which is enforced throughout this code.
+
+        Parameters
+        ----------
+        axis : 0, 1, or 2
+            The axis along which to compute the pixel size
+
+        Returns
+        -------
+        Pixel size in units of either degrees or the appropriate spectral unit
+        """
+        if axis == 0:
+            # note that self._spectral_scale is required here because wcs
+            # forces into units of m, m/s, or Hz
+            return self.wcs.pixel_scale_matrix[2,2] * self._spectral_scale
```

Comment:
@keflavich - I get a sign flip since the cube I'm testing with locally has `self.wcs.pixel_scale_matrix[2,2] < 0`. Is it safe to force an absolute value here? Or WCS isn't doing something consistently since `_, _, spectral = self._wcs.all_pix2world(xpix, ypix, zpix, 0)` returns positive values in `_pix_size`.